aten/src/ATen/native/cuda/IndexKernel.cu

```diff
@@ -249,7 +249,14 @@ void index_put_kernel_quantized_cuda(TensorIterator& iter, const IntArrayRef ind
 
     gpu_index_kernel(iter, index_size, index_stride, [inv_scale, zero_point, qmin, qmax]C10_DEVICE(char* const out_data, const char* const in_data, const int64_t offset) {
       int64_t qvalue = static_cast<int64_t>(zero_point + nearbyintf(*(float*)in_data * inv_scale));
+      // See https://github.com/pytorch/pytorch/issues/127666
+      // hip-clang std::clamp __glibcxx_assert_fail host function when building on Fedora40/gcc14
+#ifndef USE_ROCM
       qvalue = std::clamp(qvalue, qmin, qmax);
+#else
+      int64_t new_max = std::max<int64_t>(qmin, qvalue);
+      qvalue = std::min<int64_t>(qmax, new_max);
```

Comment:
Thanks for this pull request to fix the issue. It looks good logically.

Another way to implement this is as below:
return (qvalue < qmin) ? qmin : (qmax < qvalue) ? qmax : qvalue;

In average, the above should need fewer comparison(s) than the "std::max followed by std::min" work-around.

Reply:
Thanks for the suggestion.  I wasn't thinking to try and improve upon the patch.  Next time you can use the github suggest change feature and I can commit it directly from your review. :-)